sequential/seq2pat.py

```diff
@@ -458,6 +498,89 @@ def get_patterns(self, min_frequency: Num) -> List[list]:
+        """
+
+        # Shuffle sequences uniformly to make each batch have similar pattern occurrences
+        sequences, attr_to_cs = shuffle_data(self.sequences, self.attr_to_cts, self.seed)
+
+        # Get number of chunks
+        n_sequences = len(sequences)
+        num_chunks = n_sequences // self.batch_size
+        if n_sequences % self.batch_size > 0:
+            num_chunks += 1
+
+        # Run Seq2Pat in parallel on each chunk, with a lower min_frequency threshold to get a larger set of patterns
+        batch_patterns = Parallel(n_jobs=self.n_jobs, require='sharedmem')(
+            delayed(self._mining_batch)(i, sequences, attr_to_cs, min_frequency)
+            for i in range(0, num_chunks))
+
+        # Aggregate patterns from each chunk over the counts and apply min_frequency threshold
+        min_row_count = int(n_sequences * min_frequency) if isinstance(min_frequency, float) else min_frequency
+        agg_patterns = aggregate_patterns(batch_patterns, min_row_count)
+
+        return agg_patterns
+
+    def _mining_batch(self, chunk_ind, sequences, attr_to_cs, min_frequency):
+        """
+        Implementor of pattern mining on each batch
+
+        Attributes
+        ----------
+        chunk_ind: int
+           chunk index
+        sequences: List[list]
+           A list of sequences each with a list of events.
+           The event values can be all strings or all integers.
+        attr_to_cs: Dict[Attribute, Dict[str, Constraint]]
+        min_frequency: Num
+           If int, represents the minimum number of sequences (rows) a pattern should occur.
+           If float, should be (0.0, 1.0] and represents
+           the minimum percentage of sequences (rows) a pattern should occur.
+
+        Returns
+        -------
+        List[list] where each inner list represents a frequent pattern in the form
+        [event_1, event_2, event_3, ... event_n, frequency].
+        The last element is the frequency of the pattern.
+        Sequences are sored by decreasing frequency, i.e., most frequent pattern first.
+        """
+
+        # Get one batch of sequences
+        batch_sequences = sequences[chunk_ind * self.batch_size:(chunk_ind + 1) * self.batch_size]
+
+        # Create seq2pat instance for one batch
+        batch_seq2pat = Seq2Pat(batch_sequences, max_span=None, batch_size=None)
+
+        # Create constraints for one batch
+        for attr in attr_to_cs:
+            for cs in attr_to_cs[attr]:
+                old_constraint = attr_to_cs[attr][cs]
+                new_constraint = deepcopy(old_constraint)
+                new_constraint.attribute.set_values(old_constraint.attribute.values[chunk_ind * self.batch_size:
+                                                                                    (chunk_ind + 1) * self.batch_size])
+                batch_seq2pat.add_constraint(new_constraint)
+
+        # Reduce min_frequency to be min_frequency * min_frequency_df, 0 < min_frequency_df < 1
```

Comment:
Will add more comments here. So I am applying a lower min_frequency threshold in mining patterns from individual batch. The new threshold is calculated by applying a factor with value in 0-1 to min_frequency, `min_frequency * min_frequency_df`, where min_frequency is a percentage. The result will be smaller than min_frequency.

Reply:
cannot this be calculated from the given original min_freq and the batch_size.. If I am looking at %30 frequency at the whole data, and now I want 2 batches, would not that be %15 in each batch?

Reply:
You mean the row counts, not the percentage right? For row counts, yes we can calculate but I think percentage would be easier. In terms of percentage, for each batch, it will still be 30%, then the sum of two batches will become 30% of entire set.

Reply:
Ah, had the same comment above..